code/modules/codex/entries/mobs.dm

```diff
@@ -1,3 +1,65 @@
+/mob/living/carbon/Xenomorph/get_antag_info()
+	var/list/entries = SScodex.retrieve_entries_for_string(name)
+	var/datum/codex_entry/general_entry = LAZYACCESS(entries, 1)
+	if(general_entry?.antag_text)
+		return general_entry?.antag_text
+
+/mob/living/carbon/Xenomorph/get_lore_info()
+	var/list/entries = SScodex.retrieve_entries_for_string(name)
+	var/datum/codex_entry/general_entry = LAZYACCESS(entries, 1)
+	if(general_entry?.lore_text)
+		return general_entry?.lore_text
+
+/mob/living/carbon/Xenomorph/get_mechanics_info()
+	. = ..()
+	var/list/xeno_strings = list()
+
+	var/list/entries = SScodex.retrieve_entries_for_string(name)
+	var/datum/codex_entry/general_entry = LAZYACCESS(entries, 1)
+	if(general_entry?.mechanics_text)
+		xeno_strings += general_entry?.mechanics_text + "<br>"
+
+	xeno_strings += "<br><U>Basic Statistics for this Xeno are as follows</U>:"
+
+	xeno_strings += "Name: '[name]'"
+	xeno_strings += "Tier: [tier]"
+	xeno_strings += "Melee slash damage: between [xeno_caste.melee_damage_lower] and [xeno_caste.melee_damage_upper]"
+	xeno_strings += "Tackle damage: [xeno_caste.tackle_damage]"
+	switch(mob_size)
+		if(MOB_SIZE_BIG)
+			xeno_strings += "Can smash walls: Yes"
+		if(MOB_SIZE_XENO)
+			xeno_strings += "Can smash walls: No"
+	xeno_strings += "Max health: [xeno_caste.max_health]"
+	xeno_strings += "Armor: [xeno_caste.armor_deflection]"
+	xeno_strings += "Max plasma: [xeno_caste.plasma_max]"
+	xeno_strings += "Plasma gain: [xeno_caste.plasma_gain]"
+	switch(hivenumber)
+		if(XENO_HIVE_NORMAL)
+			xeno_strings += "Hive: Normal"
+		if(XENO_HIVE_CORRUPTED)
+			xeno_strings += "Hive: Corrupted"
+		if(XENO_HIVE_ALPHA)
+			xeno_strings += "Hive: Alpha"
+		if(XENO_HIVE_BETA)
+			xeno_strings += "Hive: Beta"
+		if(XENO_HIVE_ZETA)
+			xeno_strings += "Hive: Zeta"
+
+	if(length(xeno_caste.evolves_to) > 0)
```

Comment:
If length( should be enough

Reply:
removing the brackets for the if statement causes errors.

Reply:
He means `if(length(xeno_caste.evolves_to))` instead, because `length` will never return negative numbers, and you are just checking for zero here.

Reply:
```suggestion
	if(length(xeno_caste.evolves_to))
```
In suggestion easy-to-apply format.